src/webui/www/public/login.html

```diff
@@ -6,60 +6,58 @@
-                onComplete: function() {
-                    $('password').set('value', '');
-                },
-                onFailure: function(xhr) {
-                    if (xhr.responseText !== "") {
-                        $('error_msg').set('html', xhr.responseText);
-                    }
-                    else {
-                        $('error_msg').set('html', 'QBT_TR(Unable to log in, qBittorrent is probably unreachable.)QBT_TR[CONTEXT=HttpServer]');
-                    }
-                },
-                onSuccess: function(text) {
-                    if (text == "Ok.") {
-                        // Session started. Simply reload page.
-                        window.location.reload();
-                    }
-                    else {
-                        $('error_msg').set('html', 'QBT_TR(Invalid Username or Password.)QBT_TR[CONTEXT=HttpServer]');
-                    }
+            var errorMsgElement = document.getElementById('error_msg');
+
+            var xhr = new XMLHttpRequest();
+            xhr.open('POST', 'api/v2/auth/login', true);
+            xhr.setRequestHeader('Content-type', 'application/x-www-form-urlencoded; charset=utf-8');
+            xhr.addEventListener('readystatechange', function() {
+                if (xhr.readyState === 4) {  // DONE state
+                    if ((xhr.status === 200) && (xhr.responseText === "Ok."))
+                        location.reload(true);
+                    else
+                        errorMsgElement.innerHTML = 'QBT_TR(Invalid Username or Password.)QBT_TR[CONTEXT=HttpServer]';
                 }
-            }).send();
+            });
+            xhr.addEventListener('error', function() {
+                errorMsgElement.innerHTML = (xhr.responseText !== "")
+                    ? xhr.responseText
+                    : 'QBT_TR(Unable to log in, qBittorrent is probably unreachable.)QBT_TR[CONTEXT=HttpServer]';
+            });
+
+            var usernameElement = document.getElementById('username');
+            var passwordElement = document.getElementById('password');
+            var queryString = "username=" + encodeURIComponent(usernameElement.value) + "&password=" + encodeURIComponent(passwordElement.value);
+            xhr.send(queryString);
+
+            // clear the field
+            passwordElement.value = '';
         }
     </script>
 </head>
 
 <body>
+    <noscript id="noscript"><h1>QBT_TR(JavaScript Requried! You must enable JavaScript for Web UI to work properly)QBT_TR[CONTEXT=HttpServer]</h1></noscript>
     <div id="main">
         <h1>qBittorrent QBT_TR(Web UI)QBT_TR[CONTEXT=OptionsDialog]</h1>
         <div id="logo" class="col">
             <img src="images/skin/qbittorrent-tray.svg" alt="qBittorrent logo" />
         </div>
         <div id="formplace" class="col">
-            <form id="loginform">
+            <form id="loginform" method="post" onsubmit="submitLoginForm();">
```

Comment:
This form won't work with JavaScript disabled. I think we should either not render the form or change the form action to `api/v2/auth/login`.

Reply:
>change the form action to api/v2/auth/login so that the creds actually go to the right place.

yes but it still requires a page reload which I have no idea how to get it done without js.

>I think we should either not render the form 

I can try this, the only idea I can come up with is setting `display:none` to `formplace` by default and remove the property with js on DOM loaded.

Reply:
Apparently `<noscript>` is valid in `<head>` in HTML5, so we can use a solution like this: https://stackoverflow.com/a/1332058/5773436

Reply:
>Apparently <noscript> is valid in <head> in HTML5, so we can use a solution like this: https://stackoverflow.com/a/1332058/5773436

Cool! will do it.
Another offtopic question, how about moving js code to its own file? It is considered a good practice right?